### PR TITLE
[api-draft][connectors]add Assert sink to api-draft

### PR DESCRIPTION
**Pull request by lhyundeadsoul**
## Purpose of this pull request
add assert sink to api-draft branch
https://github.com/apache/incubator-seatunnel/issues/1912#issuecomment-1162595940

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by lhyundeadsoul** [comment]
PTAL @ruanwenjun Thx.

File: pom.xml [hunk 1]
```diff
@@ -433,6 +434,11 @@
                 <artifactId>commons-lang3</artifactId>
                 <version>${commons-lang3.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.apache.commons</groupId>
+                <artifactId>commons-collections4</artifactId>
+                <version>${commons-collections4.version}</version>
+            </dependency>
```
Comment (CalvinKirs) [inline pom.xml:442]: please add license files.

Comment (CalvinKirs) [reply pom.xml:442]: you can refer this https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md

Comment (lhyundeadsoul) [reply pom.xml:441]: Thanks for reviewing.  @CalvinKirs  But how to add licenses according to standard procedures? Thx again.

Comment (lhyundeadsoul) [reply pom.xml:442]: `commons-collections4-4.4.jar` is already in `known-dependencies-jdk8.txt`
`org.apache.commons:commons-collections4:4.4` is already in `seatunnel-dist/release-docs/LICENSE`.

 Is that ok?

Comment (lhyundeadsoul) [reply pom.xml:442]: > you can refer this https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md

I have 2 questions after trying add new license refering this page.
1. I got following message after the 3rd step.  Is this normal?
```
> $ python3 tools/dependencies/license.py seatunnel-dist/target/THIRD-PARTY.txt seatunnel-dist/release-docs/LICENSE true                                                                                                                                      [±assertsink4apidraft ●]

(Unknown license) jsch (com.jcraft:jsch:0.1.54 - no url defined)
(Unknown license) disruptor (com.lmax:disruptor:3.3.6 - no url defined)
(Unknown license) Commons IO (commons-io:commons-io:2.4 - http://commons.apache.org/io/)
(Unknown license) commons-net (commons-net:commons-net:3.6 - no url defined)
(Unknown license) joda-time (joda-time:joda-time:1.6 - no url defined)
(Unknown license) "Java Concurrency in Practice" book annotations (net.jcip:jcip-annotations:1.0 - http://jcip.net/)
(Unknown license) stringtemplate (org.antlr:stringtemplate:3.2.1 - no url defined)
(Unknown license) libthrift (org.apache.thrift:libthrift:0.9.0 - no url defined)
(Bouncy Castle Licence) Bouncy Castle PKIX, CMS, EAC, TSP, PKCS, OCSP, CMP, and CRMF APIs (org.bouncycastle:bcpkix-jdk15on:1.68 - http://www.bouncycastle.org/java.html)
(Bouncy Castle Licence) Bouncy Castle Provider (org.bouncycastle:bcprov-ext-jdk15on:1.68 - https://www.bouncycastle.org/java.html)
(Bouncy Castle Licence) Bouncy Castle Provider (org.bouncycastle:bcprov-jdk15on:1.68 - http://www.bouncycastle.org/java.html)
(Unknown license) xz (org.tukaani:xz:1.5 - no url defined)
(Unknown license) stax-api (stax:stax-api:1.0.1 - no url defined)
Please confirm the license by finding LICENSE file in the corresponding Jar file and maintain it in the dependency_licenses_map instance.

```
2. There is no detail instruction in 4th step. I don't know what to do for this step.

Comment (CalvinKirs) [reply pom.xml:442]: If there are no other considerations, you just need to modify `common-collections.version` to 4.2, which is already used  4.2 in our project;

Comment (lhyundeadsoul) [reply pom.xml:442]: > If there are no other considerations, you just need to modify `common-collections.version` to 4.2, which is already used 4.2 in our project;

ok, Done.

**Comment by lhyundeadsoul** [comment]
It seems a `AbstractSimpleSink` was added. I need to extend it, right?

**Comment by lhyundeadsoul** [comment]
I don't know why there are so many other's commit in my PR. so I create another PR  https://github.com/apache/incubator-seatunnel/pull/2071

**Comment by Hisoka-X** [comment]
Close this PR, change to #2071

**Pull request closed without merging**